### PR TITLE
Migrate examples to null-safety

### PR DESCRIPTION
**Pull request by luanpotter**

### CODE REVIEW

File: examples/lib/main.dart [hunk 1]
```diff
@@ -15,7 +15,7 @@ import 'stories/widgets/widgets.dart';
 
 void main() async {
   final dashbook = Dashbook(
-    title: 'Flame Example',
+    // title: 'Flame Example',
```
Comment (luanpotter) [inline examples/lib/main.dart:18]: this seems to be broken on dashbook NS

File: packages/flame/lib/src/anchor.dart [hunk 2]
```diff
@@ -87,7 +87,10 @@ class Anchor {
     } else {
       final regexp = RegExp(r'^\Anchor\(([^,]+), ([^\)]+)\)');
       final matches = regexp.firstMatch(name)?.groups([1, 2]);
-      assert(matches != null && matches.length == 2, 'Bad anchor format');
+      assert(
```
Comment (erickzanardo) [inline packages/flame/lib/src/anchor.dart:90]: Leaving a random comment to show Luan that this works fine!

**Review by erickzanardo — APPROVED** [review]
LGTM

**Review by renancaraujo — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]

File: examples/lib/main.dart (hunk 1 above)
Comment (luanpotter) [reply examples/lib/main.dart:18]: updated!

**Pull request merged**